python/tests/test_duplicate_encode.py

```diff
@@ -0,0 +1,42 @@
+import random
+from datetime import datetime
+from functools import partial
+
+import pytest
+
+import duplicate_encode
+
+INPUT_TEXT_CHUNK_SIZE = 10_000
+NUM_INPUT_TEXT_CHUNKS = 10
+
+
+@pytest.fixture(scope='module')
+def input_text():
+    random.seed(42)
+    input_text_chunk = "".join(
+        [random.choice([chr(x) for x in range(ord("0"), ord("z"))]) for _ in range(INPUT_TEXT_CHUNK_SIZE)])
+    input_text = input_text_chunk * NUM_INPUT_TEXT_CHUNKS
```

Comment:
Just catching this now, but...
1. Wouldn't this range be ASCII `0` to `y` because the second parameter of range is exclusive?
2. By repeating the input, aren't we eliminating the case where a character shows up once in the string? (actually, this probably doesn't matter... the input length is so much longer than the number of possible characters that there's a very low chance you'll get something unique anyways)

Reply:
We can solve both of these problems by adding a `z` at the beginning of the input, right? Otherwise, we could do what's suggested above and just use a static 100k file as the input chunk. Preference?

Reply:
I resolved this in the rust version by making the range inclusive, and by generating the whole input randomly (rather than by generating part of the input and repeating it). I'm not attached to this method though, and it's still unlikely that we'll get unique characters like this, so I'm leaning towards a file as input.